pandas/_libs/tslibs/np_datetime.pxd

```diff
@@ -71,6 +71,8 @@ cdef extern from "src/datetime/np_datetime.h":
                                              pandas_timedeltastruct *result
                                              ) nogil
 
+    int64_t hash_datetime_from_struct(npy_datetimestruct* dts)
```

Comment:
```suggestion
    Py_hash_t hash_datetime_from_struct(npy_datetimestruct* dts) except -1
```
might have to import that, but `Py_hash_t` can be 32bit on 32bit platforms and using the wrong thing might lead to weird problems.

Also, we return -1 on error, which you need to tell cython about.